### DISCUSSION TITLE
[Android] Bone rotation

### DISCUSSION THREAD

**Discussion by S8enson** [google/filament#5895] [Q&A] [answered]
Hi,

I currently have the same issue as #4728 had, as their file has been deleted I am unable to see how they solved it. I understand that renderableManager should be used, but am unsure how to do this. I am using [this model](https://drive.google.com/file/d/1o0io8Vr6qZQRtPzt7loqM2fdsI5YIrY4/view?usp=sharing) and want to bend it at the base of the spine01 bone. 

Thanks

Edit: I am wondering is it possible to rotate in the same way it is done when the spine transform node is selected as such 
![Screen Shot 2022-08-08 at 6 13 45 PM](https://user-images.githubusercontent.com/67305315/183351383-d0b37be1-900a-45f2-ac5b-026d5011dddf.png)
And if so how do I access/rotate the node

Edit 2: Here is my fixed code for anyone with the same issue
`
private val frameCallback = object : Choreographer.FrameCallback {
        private val startTime = System.nanoTime()
        override fun doFrame(currentTime: Long) {

            val seconds = (currentTime - startTime).toDouble() / 1_000_000_000
            choreographer.postFrameCallback(this)

            modelViewer.asset?.getFirstEntityByName("spine01")?.let { entity ->
                val defaultTransform = entity.getTransform()
                val transform = defaultTransform * rotation(
                    Float3(1.0f, 0.0f, 0.0f),
                    sin(seconds.toFloat()) * 2
                )
                entity.setTransform(transform)
            }
            modelViewer.animator?.updateBoneMatrices()

            modelViewer.render(currentTime)
        }
    }

    private fun Int.getTransform(): Mat4 {
        val tm = modelViewer.engine.transformManager
        val arr = FloatArray(16)
        tm.getTransform(tm.getInstance(this), arr)
        return Mat4.of(*arr)
    }

    private fun Int.setTransform(mat: Mat4) {
        val tm = modelViewer.engine.transformManager
        tm.setTransform(tm.getInstance(this), mat.toFloatArray())
    }
`

**Comment by yongwookim** [comment, marked as answer]
when you done loading your model, use transformManager's getTransform for each node to get initial transformation. Make sure you use it as basis of each node's rotation(transformation) to be applied to using transform manager's setTransform.

Personally, I tried to use renderable manager's setBonesAsQuaternions for each node's rotation and was unable to solve issue with child node not following parent node's rotation.

**Reply by S8enson** [reply]
Do you have an example I could see please? Struggling to understand

Edit: managed to figure it out. Thank you so much! :)
